include/jemalloc/internal/arena_structs_b.h

```diff
@@ -123,6 +123,13 @@ struct arena_s {
 	 */
 	unsigned		nthreads[2];
 
+	/*
+	 * When percpu_arena is enabled, to amortize the cost of reading /
+	 * updating the current CPU id, track the most recent thread accessing
+	 * this arena, and only read CPU if there is a mismatch.
+	 */
+	tsdn_t		*last_thd;
```

Comment:
Oh if it's a new thread, tsd->arena will be NULL which triggers arena_choose_hard (that does read cpu). If it happens to be on the same core & arena, it should be fine too, and even the last_thd mismatch will be dodged.